internal/liqonet/route-operator/symmetricRoutingOperator.go

```diff
@@ -0,0 +1,165 @@
+package routeoperator
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"sync"
+
+	k8sApiErrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"golang.org/x/sys/unix"
+
+	liqoerrors "github.com/liqotech/liqo/pkg/liqonet/errors"
+	"github.com/liqotech/liqo/pkg/liqonet/routing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/kubernetes/pkg/api/v1/pod"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	"github.com/liqotech/liqo/pkg/liqonet/overlay"
+)
+
+// SymmetricRoutingController reconciles pods objects, in our case all the existing pods.
+type SymmetricRoutingController struct {
+	client.Client
+	vxlanDev       overlay.VxlanDevice
+	nodeName       string
+	routingTableID int
+	nodesLock      *sync.RWMutex
+	vxlanNodes     map[string]string
+	routes         map[string]string
+}
+
+// Reconcile for a given pod, based on the node where the pod is running, configures a route to send the traffic
+// through the overlay network. The route is used only for the traffic generated on remote clusters with
+// destination address a pod running on the local cluster. The traffic for local pods should be handled by
+// the local CNI but if the reverse path filtering is in strict mode (default) the traffic coming from a
+// network interface where the source address is not routable by the same interface is dropped. To solve the
+// issue it requires to set the reverse path filtering to loose mode, but some times it is not possible. To
+// overcome the problem all the traffic sent to or coming from a peering cluster has to be routed using the
+// overlay network.
+func (src *SymmetricRoutingController) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
```

Comment:
Question: Is resync configured? How often?

Reply:
Resync is not configured. The operator uses a custom routing table. The idea is to avoid the possibility that external processes mess with the configuration. The operator handles all the existing pods on the cluster, I would prefer to not have the rysync option.

Reply:
Yes, I was thinking of the scenario when a route is added for the creation of one pod. Then, if when the pod is deleted, the operator is not available: how can we remove the "orphaned" route? Does the operator flush the entire table at each restart?

Reply:
Yes, that is right, the custom routing table is flushed at exit time by the operator.